CONTRIBUTING.md

```diff
@@ -1,4 +1,44 @@
+
+Testing Models:
+
+All models should have unit tests that can be found in the `tests/unit` directory. Each model should have tests that test the inherited methods from the BaseModel to ensure that nothing crazy is inadvertently done to change the behavior of the methods that the BaseModel provides. These are easily implemented using the base_interface_test file. Finally, there should be a test for every public method that is defined in the model. This would be the json method or any other method defined in the file that is directly used outside of the Models class.
+
+Schemas:
+
+Marshmallow Schemas are used primarily for input validation and deserialization. And eventually serialization. Schemas receive the data that the client sends the back-end and validates that data before the data is inserted into the database. They can also describe how the data is serialized before sending data to the client. Schemas can be found in the `schemas` directory.
+
+Testing Schemas:
+
+All schemas should have unit tests. Primarily validation tests. This app uses flask-marshmallow which provides an auto-schema that infers some basic validations based off of the table definition in the Models class. Any additional validations defined in the schema must be tested Schema tests can be found in the `tests/schemas` directory. Deserialization should also be tested here when used. Serialization is currently not used and does not need to be tested here at this time. It's possible that testing serialization will take place elsewhere.
+
+Resources:
+
+A resource main job is to coordinate a response for the request. If data is provided the resource will send that data to the schema for validation and deserialization. The resource will communicate with the model to query for data or insert/update a tables row in the database. Finally the resource will send a response back to the client. Resources can be found in the `resources` directory.
+
+Testing Resources:
+
+Each resource will usually have one test for each action (GET, POST, DELETE, etc...). When the Models and Schemas have unit tests, and when the resource uses the models and schemas appropriately. Then **generally** only a successful reponse (The Happy Path) needs to be tested. As all other responses that can occurr happen elsewhere and are already under test. Such as validation errors, or a database row cannot be found. Errors such as these need not be tested as they're already built into the architecture of the app and happen automatically as long as the schemas are used along with the appropriate methods defined in the BaseModel. Tests for the resources can be found in the `tests/integration` directory.
```

Comment:
occur/occurr (sp?)

Reply:
"A resource's main job is to coordinate a response for the request." (add 's to resource)

"When the Models and Schemas have unit tests, and when the resource uses the models and schemas appropriately, then **generally** only a successful response (The Happy Path) needs to be tested." (combine the two sentences and fix 'reponse' spelling)